File: Labs/Lab5/Lab5Test.java
import org.junit.Test;

import static org.junit.Assert.*;

public class Lab5Test {

    @Test
    public void test1() {
        TreeNode t = new TreeNode(4);
        t.left = new TreeNode(3);
        t.left.left = new TreeNode(1);
        t.right = new TreeNode(8);
        t.right.left = new TreeNode(5);
        t.right.right = new TreeNode(9);
        assertEquals(true, TreeNode.isValidBST(t));
    }
    @Test
    public void test2() {
        TreeNode t = new TreeNode(1);
        t.left = new TreeNode(2);
        t.left.left = new TreeNode(3);
        t.left.right = new TreeNode(4);
        t.right = new TreeNode(8);
        t.right.left = new TreeNode(5);
        t.right.right = new TreeNode(6);
        assertEquals(false, TreeNode.isValidBST(t));
    }
}

File: Labs/Lab5/lab5Test.py
import unittest
from TreeNode import *

class MyTestCase(unittest.TestCase):
    def test_lab5(self):
        t = TreeNode(4)
        t.left = TreeNode(3)
        t.left.left = TreeNode(1)
        t.right = TreeNode(8)
        t.right.left = TreeNode(5)
        t.right.right = TreeNode(9)
        self.assertEqual(True, t.isValidBST(t))


if __name__ == '__main__':
    unittest.main()
